community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -86,13 +86,28 @@
         // access to these is guarded by synchronized blocks
         private int readCount;
         private int writeCount;
-        private boolean movedOn;
+        private int requests = 0;
```

Comment:
it's an active request that we have currently for this `RWLock` we need to track this to be able to know if it is safe to delete this RWLock or we already have somebody interesting in it - in that case we need to keep it.